app/code/Magento/Sales/Model/Service/OrderService.php

```diff
@@ -87,7 +87,8 @@ public function __construct(
     public function cancel($id)
     {
         $order = $this->orderRepository->get($id);
-        if ((bool)$order->cancel()) {
+        if ((bool)$order->canCancel()) {
```

Comment:
Please remove `(bool)` casting, it is useless within `if` anyway.

Reply:
fixed